pkg/application/ecosystemConfigUseCase.go

```diff
@@ -78,102 +84,132 @@ func (useCase *EcosystemConfigUseCase) ApplyConfig(ctx context.Context, blueprin
+		errs = append(errs, err)
 	}
 
 	entryDiffsToRemove := globalConfigDiffsByAction[domain.ConfigActionRemove]
-	var keysToDelete = make([]common.GlobalConfigKey, 0, len(entryDiffsToRemove))
 	for _, diff := range entryDiffsToRemove {
-		keysToDelete = append(keysToDelete, diff.Key)
+		updatedEntries = updatedEntries.Delete(diff.Key)
 	}
 
-	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.globalConfigRepository.SaveAll))
-	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.globalConfigRepository.DeleteAllByKeys))
+	if len(entryDiffsToSet) != 0 || len(entryDiffsToRemove) != 0 {
+		_, err = useCase.globalConfigRepository.Update(ctx, config.GlobalConfig{Config: updatedEntries})
+		errs = append(errs, err)
+	}
 
 	return errors.Join(errs...)
 }
 
-func (useCase *EcosystemConfigUseCase) applyDoguConfigDiffs(ctx context.Context, doguConfigDiffsByAction map[domain.ConfigAction]domain.DoguConfigDiffs) error {
+func (useCase *EcosystemConfigUseCase) applyDoguConfigDiffs(
+	ctx context.Context,
+	diffsByDogu map[common.SimpleDoguName]domain.CombinedDoguConfigDiffs,
+) error {
 	var errs []error
-	var entriesToSet []*ecosystem.DoguConfigEntry
-	var keysToDelete []common.DoguConfigKey
 
-	for _, diff := range doguConfigDiffsByAction[domain.ConfigActionSet] {
-		entry := &ecosystem.DoguConfigEntry{
-			Key:   common.DoguConfigKey{DoguName: diff.Key.DoguName, Key: diff.Key.Key},
-			Value: common.DoguConfigValue(diff.Expected.Value),
+	var doguNames []common.SimpleDoguName
+	for dogu, diff := range diffsByDogu {
+		if diff.DoguConfigDiff.HasChangesForDogu(dogu) {
+			doguNames = append(doguNames, dogu)
 		}
-		entriesToSet = append(entriesToSet, entry)
 	}
+	slices.Sort(doguNames)
 
-	for _, diff := range doguConfigDiffsByAction[domain.ConfigActionRemove] {
-		keysToDelete = append(keysToDelete, common.DoguConfigKey{DoguName: diff.Key.DoguName, Key: diff.Key.Key})
+	if len(doguNames) == 0 {
+		// no changes to dogu config needed
+		return nil
 	}
 
-	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.doguConfigRepository.SaveAll))
-	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.doguConfigRepository.DeleteAllByKeys))
+	configByDogu, err := useCase.doguConfigRepository.GetAll(ctx, doguNames)
+	if err != nil {
+		return err
+	}
+
+	for _, dogu := range doguNames {
+		updatedConfig, applyError := useCase.applyDiff(configByDogu[dogu], diffsByDogu[dogu].DoguConfigDiff)
+		if applyError != nil {
+			return applyError
+		}
+		_, applyError = useCase.doguConfigRepository.Update(ctx, config.DoguConfig{
+			DoguName: dogu,
+			Config:   updatedConfig,
+		})
+		if applyError != nil {
+			return applyError
+		}
+	}
 
 	return errors.Join(errs...)
 }
 
-func (useCase *EcosystemConfigUseCase) applySensitiveDoguConfigDiffs(ctx context.Context, sensitiveDoguConfigDiffsByAction map[domain.ConfigAction]domain.SensitiveDoguConfigDiffs) error {
+func (useCase *EcosystemConfigUseCase) applySensitiveDoguConfigDiffs(
```

Comment:
🚫 **[checkstyle]** <dupl> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>154-193 lines are duplicate of `pkg/application/ecosystemConfigUseCase.go:113-152`
<!-- __reviewdog__:ChBjOGFlMmFmYTViZjgwNDExEgpjaGVja3N0eWxl -->

Reply:
I will try to remove this duplicate by refactoring the doguDiff, so that we can handle sensitiveDoguConfig like normal doguConfig. After removing the encryption, the only difference at this point is, the used repository.